### PR TITLE
Document automator

### LINKED ISSUE #2786: Create README and test steps for automator
Improve the documentation for [automator](https://github.com/istio/test-infra/tree/master/tools/automator) by providing a comprehensive README and a set of steps to debug and/or test locally.

### LINKED ISSUE #2798: What is "Automator" ?
Hello,

I see istio has interesting automation going on which updates the docker versions after a successful build. Could someone please offer some info/references to how it's configured and where I can learn more about it?

### PR DESCRIPTION
**Pull request by clarketm**
fix #2798, fix #2786

### CODE REVIEW

File: tools/automator/README.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+1. [Bash 4.0+] interpreter.
+
+1. The following third-party tools must be available in the execution environment:
+   - [`curl`]
+   - [`getopt`]
+   - [`git`]
+   - [`jq`]
+   - [`pr-creator`]
+   - [`realpath`]
+
```
Comment (ericvn) [inline tools/automator/README.md:20]: I'm not sure if we want to document anything additional, but I also needed to install `gnu-getopt` on the Mac the get things to work when running locally. The normal `getopt` did not work.

File: tools/automator/README.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+[prow jobs]: https://github.com/istio/test-infra/blob/de701f31e850f48fc035a5ec383c2595661dcc7b/prow/cluster/jobs/istio/istio.io/istio.istio.io.master.gen.yaml#L3-L46
+[github access token]: https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token
+[bash 4.0+]: https://www.tldp.org/LDP/abs/html/bashver4.html
+[git diff]: https://git-scm.com/docs/git-diff
+[`curl`]: https://curl.haxx.se/
+[`getopt`]: https://linux.die.net/man/1/getopt
+[`git`]: https://git-scm.com/
+[`jq`]: https://github.com/stedolan/jq
+[`pr-creator`]: https://github.com/kubernetes/test-infra/tree/master/robots/pr-creator
+[`reealpath`]: https://linux.die.net/man/1/realpath
```
Comment (ericvn) [inline tools/automator/README.md:116]: For Mac, I used: `brew install bash` (installs v5) and `brew install gnu-getopt`

Comment (ericvn) [reply tools/automator/README.md:116]: Another Mac issue is you get an error (sample command)
```
../test-infra/tools/automator/automator.sh --org istio --repo istio --dry-run --modifier=update_proxy_dep --cmd='bin/update_proxy.sh $AUTOMATOR_SHA'
/var/folders/jw/qff4nk3s6f97x5wht80t3w0r0000gn/T/ci-XXXXXXXXXX.dpEQYrKP ~/work/go/src/istio.io/proxy
date: illegal option -- I
usage: date [-jnRu] [-d dst] [-r seconds] [-t west] [-v[+|-]val[ymwdHMS]] ...
            [-f fmt date | [[[mm]dd]HH]MM[[cc]yy][.ss]] [+format]
```
again due to a Mac inconsistency. To fix, one needs to `brew install coreutils` and then alias _date_ to _gdate_.

**Review by ericvn — APPROVED** [review]
For now, we can leave the Mac changes out as I'm not sure how much they are needed.

File: tools/automator/README.md (hunk 2 above)
Comment (clarketm) [reply tools/automator/README.md:116]: I will follow-up and add the Mac-specific tools to the README as well.

Comment (clarketm) [reply tools/automator/README.md:116]: https://github.com/istio/test-infra/pull/2881

**Pull request merged**